pygen/pygen_src/test/riscv_instr_cov_test.py

```diff
@@ -0,0 +1,293 @@
+        self.rd_special_value = self.get_operand_special_value(self.rd_value)
+        self.rs2_special_value = self.get_operand_special_value(self.rs2_value)
+        self.rs3_special_value = self.get_operand_special_value(self.rs3_value)
+        if (self.format != riscv_instr_format_t.R_FORMAT and
+            self.format != riscv_instr_format_t.CR_FORMAT):
+            self.imm_special_value = self.get_imm_special_val(self.imm)
+        if self.category in [riscv_instr_category_t.COMPARE,
+                            riscv_instr_category_t.BRANCH]:
+            self.compare_result = self.get_compare_result()
+        if self.category in [riscv_instr_category_t.LOAD,
+                             riscv_instr_category_t.STORE]:
+            self.mem_addr = self.rs1_value + self.imm
+            self.unaligned_mem_access = self.is_unaligned_mem_access()
+            if self.unaligned_mem_access:
+                logging.info("Unaligned: {}, mem_addr: {}".format(
+                self.instr, self.mem_addr))
+        if self.category == riscv_instr_category_t.LOGICAL:
+            self.logical_similarity = self.get_logical_similarity()
+        if self.category == riscv_instr_category_t.BRANCH:
+            self.branch_hit = self.is_branch_hit()
+        #TODO: string > enumeration
+        if self.instr in ["DIV", "DIVU", "REM", "REMU", "DIVW", "DIVUW", 
+                          "REMW", "REMUW"]:
+            self.div_result = self.get_div_result()
+
+    def get_operand_sign(self, value):
+        #TODO: cast value to vsc.bit_t
+        if value[0]:
+            return operand_sign_e["NEGATIVE"]
+        else:
+            return operand_sign_e["POSITIVE"]
+
+    def is_unaligned_mem_access(self):
+        #TODO: string > enumeration
+        if (self.instr in ["LWU", "LD", "SD", "C_LD", "C_SD"] and
+            self.mem_addr % 8 != 0):
+            return True
+        elif (self.instr in ["LW", "SW", "C_LW", "C_SW"] and
+              self.mem_addr % 4 != 0):
+            return True
+        elif (self.instr in ["LH", "LHU", "SH"] and
+              self.mem_addr % 2 != 0):
+            return True
+        return False
+
+    def get_imm_sign(self):
+        pass
+
+    def get_operand_special_value(self):
+        pass
+
+    def get_imm_special_val(self):
+        pass
+
+    def get_compare_result(self):
+        pass
+
+    def get_logical_similarity(self):
+        pass
+
+    def is_branch_hit(self):
+        pass
+
+    def get_div_result(self):
+        pass
+
+    def update_src_regs(self, operands):
+        pass
+
+    def update_dst_regs(self, reg_name, val_str):
+        pass
+
+class riscv_instr_cov_test(argv):
+    """ Main class for applying the functional coverage test """
+    def __init__(self):
+        self.trace = {}
+        self.entry_cnt, self.total_entry_cnt, self.skipped_cnt, \
+        self.unexpected_illegal_instr_cnt = 0, 0, 0, 0
+
+    def run_phase(self):
+        logging.info("{} CSV trace files to be "
```

Comment:
should we short-circuit and `sys.exit` right away if argv is empty?

Reply:
Yup, makes sense.